source/obstacle/Obstacle.hx

```diff
@@ -1,57 +1,81 @@
 package obstacle;
 
+import flixel.FlxBasic;
 import flixel.FlxG;
 import flixel.FlxSprite;
 
-class Obstacle extends FlxSprite
-{
-	public function new(?parameters:ObstacleParameters)
-	{
+class Obstacle extends FlxSprite {
+	public function new(?parameters:ObstacleParameters) {
 		super();
 
 		immovable = true;
 
-		if (parameters != null)
-		{
+		if (parameters != null) {
+			// this gives the obstacle its init values
 			init(parameters);
 		}
 	}
 
-	public function init(parameters:ObstacleParameters)
-	{
+	public function init(parameters:ObstacleParameters) {
+		// revive the obstace to be drawn again
 		super.reset(parameters.x, parameters.y);
 
+		// set the velocity to whatever the speed is but neg
+		var upOrDown = FlxG.random.int(0, 1);
 		velocity.x = -parameters.speed;
 
+		if (upOrDown == 1) {
+			velocity.y = -(FlxG.random.float(parameters.speed / 2, parameters.speed));
+		} else {
+			velocity.y = (FlxG.random.float(parameters.speed / 2, parameters.speed));
+		}
```

Comment:
It makes sense to me that this sort of behavior doesn't play nice with the way parameters and variation works right now, but it might be nice to change the system so that it better accounts for the sort of thing you want to do.

I know that code is pretty messy, though, so it's not a requirement to do something like that.